### DISCUSSION TITLE
Unable to communicate with MIDI hardware

### DISCUSSION THREAD

**Discussion by ttshaw1** [jazz-soft/web-midi#6] [Q&A]
Hi,

I'm running Firefox 89.0 on Linux and am trying to get MIDI working. I've installed the Web MIDI API and Jazz-MIDI extensions, as well as Jazz-Plugin. 

My goal is to be able to use Firefox to control my guitar pedal through [this project ](https://github.com/g200kg/zoom-ms-utility), which allows effects to be set up in a browser page and the pedal programmed via MIDI; as of right now it seems that Firefox won't communicate with MIDI devices. The same is true when checking [this page](https://github.com/g200kg/zoom-ms-utility) with either my pedal or MIDI controller keyboard plugged in. The MIDI Data Log only says "Initializing MIDI..."

I can see the extensions listed in Firefox, but don't see the plugin listed. I had a bit of a hard time installing the plugin. Running the archive script returned only 
"Verifying archive integrity...  100%   All good.
Uncompressing Jazz-Plugin  100%  "

Eventually I set the script to extract to a specific folder and ran install.sh in that folder, and was then able to verify that the files had been copied to the right folders (e.g. /usr/lib/mozilla/plugins/). But the fact that I still don't see the plugin in Firefox makes me wonder if it installed correctly.

Do you have any advice?

**Comment by jazz-soft** [comment]
Thanks for asking!
Here is what the install script does:

sudo cp ./64/jazz-chrome /usr/lib/mozilla/plugins/
sudo chmod 755 /usr/lib/mozilla/plugins/jazz-chrome
sudo cp ./firefox/com.jazz_soft.jazz_midi.json /usr/lib/mozilla/native-messaging-hosts/
sudo chmod 644 /usr/lib/mozilla/native-messaging-hosts/com.jazz_soft.jazz_midi.json

please make sure the paths are correct in com.jazz_soft.jazz_midi.json

Jazz-Plugin.so is not used in modern browsers and I should remove it from the installer.

**Reply by ttshaw1** [reply]
>please make sure the paths are correct in com.jazz_soft.jazz_midi.json

com.jazz_soft.jazz_midi.json was copied to the expected location, and the "path" specified in "path": "/usr/lib/mozilla/plugins/jazz-chrome" is pointing to jazz-chrome as expected. Is that what you mean?
